skrub/tests/test_table_vectorizer.py

```diff
@@ -642,6 +643,120 @@ def test__infer_date_format() -> None:
+    # transformation (albeit, not necessarily in the same order) as the
+    # TableVectorizer.
+    if len(specific_transformers) == 2:
+        # Unnamed assignment
+        specific_transformers: tuple[object, list[str]]
+        transformer, columns = specific_transformers
+        default_table_vectorizer_assignment = [
+            (transformer, columns)
+            for _, transformer, columns in TableVectorizer()
+            .fit(X.drop(columns=columns))
+            .transformers
+        ]
+        ct = make_column_transformer(
+            *default_table_vectorizer_assignment, specific_transformers
+        )
+        X_enc_ct = ct.fit_transform(X)
+    elif len(specific_transformers) == 3:
+        # Named assignment
+        specific_transformers: tuple[str, object, list[str]]
+        name, transformer, columns = specific_transformers
+        default_table_vectorizer_assignment = (
+            TableVectorizer().fit(X.drop(columns=columns)).transformers
+        )
+        # Assert the name is used in the assignment
+        assert name in tv.named_transformers_
+        ct = ColumnTransformer(
+            transformers=default_table_vectorizer_assignment + [specific_transformers],
+        )
+        X_enc_ct = ct.fit_transform(X)
+
+    assert X_enc_tv.shape == X_enc_ct.shape
+
+    # Assert the output is the same.
+    # This comparison works for arrays with possibly different order,
+    # which is a specificity of the output of the ColumnTransformer ; see
+    # the TableVectorizer "Notes" section.
+    for row_index in range(X_enc_tv.shape[0]):
+        c1 = np.convolve(X_enc_tv[row_index], X_enc_tv[row_index], "valid")[0]
+        c2 = np.convolve(X_enc_tv[row_index], X_enc_ct[row_index], "valid")[0]
+        if pd.isna(c1) or pd.isna(c2):
+            continue
+        assert np.isclose(c1, c2)
+
+
+def test_specific_transformers_unexpected_behavior():
+    X = _get_clean_dataframe()
+
+    # Test that using tuple lengths other than 2 or 3 raises an error
+    with pytest.raises(TypeError):
+        assert TableVectorizer(specific_transformers=[(StandardScaler())]).fit(X)
+        assert TableVectorizer(
+            specific_transformers=[("dummy", StandardScaler(), ["float"], 1)]
+        ).fit(X)
+
+    # Test that using mixed length tuples raises an error
+    with pytest.raises(TypeError):
+        assert TableVectorizer(
+            specific_transformers=[
+                (StandardScaler(), ["float"]),
+                ("dummy", StandardScaler(), ["float"]),
+            ]
+        ).fit(X)
+
+
+@pytest.mark.parametrize(
+    "pipeline",
+    [
+        TableVectorizer(),
+        TableVectorizer(
+            specific_transformers=[
+                (MinHashEncoder(), ["cat1", "cat2"]),
+            ],
+        ),
+        TableVectorizer(
+            low_card_cat_transformer=MinHashEncoder(),
+        ),
+    ],
+)
+def test_deterministic(pipeline) -> None:
```

Comment:
I hope that this test isn't too long :p

Reply:
No, it's very fast (~400ms on my machine) since it works on a synthetic dataset which is very small (the `_get_dirty_dataframe`).